gnomad_constraint/constraint/constraint_pipeline.py

```diff
@@ -0,0 +1,63 @@
+# noqa: D100
+
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad.resources.grch38.reference_data import vep_context
+from gnomad.resources.grch37.gnomad import public_release
+from gnomad_constraint.utils.constraint_basics import (
+    add_vep_context_annotations,
+    prepare_ht_for_constraint_calculations,
+)
+
+from gnomad_constraint.resources.resource_utils import get_processed_ht_path
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("constraint_pipeline")
+logger.setLevel(logging.INFO)
+
+
+def main(args):
+    """Execute the constraint pipeline."""
```

Comment:
It could help to add a try, finally block. most of the code goes under the try. You can add copy_log (https://hail.is/docs/0.2/utils/index.html#hail.utils.copy_log) under the finally to save the log in case there's issues running the pipeline. See here for example: https://github.com/broadinstitute/gnomad_qc/blob/8ec2f0d3a20e6d466bd6ee0858d30d4cbde9ba54/gnomad_qc/v4/sample_qc/hard_filters.py#L255